br/pkg/streamhelper/advancer.go

```diff
@@ -424,7 +424,15 @@ func (c *CheckpointAdvancer) onTaskEvent(ctx context.Context, e TaskEvent) error
 		c.task = e.Info
 		c.taskRange = spans.Collapse(len(e.Ranges), func(i int) kv.KeyRange { return e.Ranges[i] })
 		c.setCheckpoints(spans.Sorted(spans.NewFullWith(e.Ranges, 0)))
-		c.lastCheckpoint = newCheckpointWithTS(e.Info.StartTs)
+		globalCheckpointTs, ok := c.env.GetGlobalCheckpointForTask(ctx, c.task.Name)
+		if ok != nil {
+			log.Warn("Advancer failed to get global checkpoint")
+			globalCheckpointTs = 0
+		}
+		if globalCheckpointTs < c.task.StartTs {
+			globalCheckpointTs = c.task.StartTs
+		}
+		c.lastCheckpoint = newCheckpointWithTS(globalCheckpointTs)
 		p, err := c.env.BlockGCUntil(ctx, c.task.StartTs)
```

Comment:
This BlockGCUntil should be `globalCheckpointTS` too.

Reply:
done